commands/common/cmd_api.go

```diff
@@ -0,0 +1,139 @@
+}
+
+func apiError(status int, message string, args ...any) *ApiError {
+	return &ApiError{
+		StatusCode: status,
+		Message:    fmt.Sprintf(message, args...),
+	}
+}
+
+type ApiCallParams struct {
+	Method      string
+	ServerUrl   string
+	ServerToken string
+	Body        []byte
+	Query       map[string]string
+	Path        []string
+	ProjectKey  string
+	ApiVersion  apiVersion
+	OkStatuses  []int
+	OnContent   ApiContentHandler
+}
+
+func CallWorkerApi(c model.IntFlagProvider, params ApiCallParams) error {
+	timeout, err := model.GetTimeoutParameter(c)
+	if err != nil {
+		return apiError(http.StatusInternalServerError, "%+v", err)
+	}
+
+	apiEndpoint := fmt.Sprintf("%sworker/api/v%d/%s", utils.AddTrailingSlashIfNeeded(params.ServerUrl), params.ApiVersion+1, strings.Join(params.Path, "/"))
+
+	if params.Query != nil || params.ProjectKey != "" {
+		var query string
+		for key, value := range params.Query {
+			query = addToQuery(query, key, value)
+		}
+		if params.ProjectKey != "" {
+			query = addToQuery(query, "projectKey", params.ProjectKey)
+		}
+		if query != "" {
+			apiEndpoint += "?" + query
+		}
+	}
+
+	reqCtx, cancelReq := context.WithTimeout(context.Background(), timeout)
+	defer cancelReq()
+
+	var bodyReader io.Reader
+	if params.Body != nil {
+		bodyReader = bytes.NewBuffer(params.Body)
+	}
+
+	req, err := http.NewRequestWithContext(reqCtx, params.Method, apiEndpoint, bodyReader)
+	if err != nil {
+		return apiError(http.StatusInternalServerError, "failed to create request: %+v", err)
+	}
+
+	req.Header.Add("Authorization", "Bearer "+strings.TrimSpace(params.ServerToken))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("User-Agent", coreutils.GetCliUserAgent())
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return apiError(http.StatusRequestTimeout, "request timed out after %s", timeout)
+		}
+		return apiError(http.StatusInternalServerError, "unexpected error: %+v", err)
+	}
+
+	if slices.Index(params.OkStatuses, res.StatusCode) == -1 {
+		// We the response contains json content, we will print it
+		_ = processApiResponse(res, printJsonOrLogError)
+		return apiError(res.StatusCode, "command %s %s returned an unexpected status code %d", params.Method, apiEndpoint, res.StatusCode)
+	}
+
+	return processApiResponse(res, params.OnContent)
+}
+
+func processApiResponse(res *http.Response, doWithContent func(content []byte) error) error {
+	var err error
+	var responseBytes []byte
+
+	defer CloseQuietly(res.Body)
+
+	if res.ContentLength > 0 || res.ContentLength == -1 {
+		responseBytes, err = io.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, _ = io.Copy(io.Discard, res.Body)
+	}
+
+	if doWithContent == nil {
+		return nil
+	}
+
+	return doWithContent(responseBytes)
+}
+
+func addToQuery(q string, key, val string) string {
+	if q != "" {
+		q += "&"
+	}
+	return fmt.Sprintf("%s%s=%s", q, key, url.QueryEscape(val))
```

Comment:
Use string concatenation instead of `Sprintf`:
```suggestion
	return q+key+"="+url.QueryEscape(val)
```